src/components/create-guild/CreateGuildForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { Center, FormControl, FormLabel, Input, Stack } from "@chakra-ui/react"
+import { Schemas } from "@guildxyz/types"
+import Color from "color"
+import ColorThief from "colorthief/dist/color-thief.mjs"
+import useUser from "components/[guild]/hooks/useUser"
+import Card from "components/common/Card"
+import FormErrorMessage from "components/common/FormErrorMessage"
+import usePinata from "hooks/usePinata"
+import { useEffect } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+import CreateGuildButton from "./CreateGuildButton"
+import IconSelector from "./IconSelector"
+import Name from "./Name"
+
+export type CreateGuildFormType = Pick<
+  Schemas["GuildCreationPayload"],
+  "name" | "imageUrl" | "contacts" | "theme"
+>
+
+const getColorByImage = (imageUrl: string) =>
```

Comment:
I think it'd be nice to move it into a util function, and delete useSetImageAndNameFromPlatformData as it's not used anymore